### PR TITLE
feat(prefer-screen-queries): add prefer-screen-queries

### LINKED ISSUE #95: RFC: new rule to prefer the usage of screen
It's preferred to use `screen` when using the query functions.

```js
// bad
const component = render()
component.getBy()

// good
screen.getBy()
```

For a first implementation we can verify that the queries used have `screen` as callee.
This doesn't take the following into account, but it makes the rule simpler.

```
const screen = render()
screen.getBy
```

### PR DESCRIPTION
**Pull request by timdeschryver**
Closes #95

### CODE REVIEW

File: lib/utils.js [hunk 1]
```diff
@@ -44,8 +44,8 @@ const ASYNC_QUERIES_COMBINATIONS = combineQueries(
 );
 
 const ALL_QUERIES_COMBINATIONS = [
-  SYNC_QUERIES_COMBINATIONS,
-  ASYNC_QUERIES_COMBINATIONS,
+  ...SYNC_QUERIES_COMBINATIONS,
+  ...ASYNC_QUERIES_COMBINATIONS,
```
Comment (timdeschryver) [inline lib/utils.js]: I hope this is OK (`ALL_QUERIES_COMBINATIONS ` wasn't used until now)

File: docs/rules/prefer-screen-queries.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+const utils = render(<Component />);
+utils.getByText('foo');
+```
+
+Examples of **correct** code for this rule:
+
+```js
+import { screen } from '@testing-library/any-framework';
+
```
Comment (Belco90) [inline docs/rules/prefer-screen-queries.md]: Can you add here the render line to have a full example?

```suggestion
render(<Component />);
```

File: lib/utils.js (hunk 1 above)
Comment (Belco90) [reply lib/utils.js]: Damn 🤦‍♂️ Thanks for this fix!

**Comment by Belco90** [comment]
I edited to PR to point it to `v3` branch and now you get some conflicts. Sorry about that.

**Comment by timdeschryver** [comment]
No problem! Should be good now, thanks for the review

**Pull request merged**